Move cache to repo

mynomoto opened the pull request:
This avoids creating the cache on each project

mynomoto committed:
Move cache to repo

This avoids creating the cache on each project

borkdude commented:
Perhaps we should rename `.work` to `.deps-infer` and document that it will write garbage into your m2 folder? I'm not sure if everybody will like this default behavior so we should probably also provide an option to override the directory.

mynomoto committed:
Update README, fix help option, allow custom cache-dir

mynomoto commented:
> Perhaps we should rename `.work` to `.deps-infer` and document that it will write garbage into your m2 folder? I'm not sure if everybody will like this default behavior so we should probably also provide an option to override the directory.

I did the rename, added the option to change the cache-dir path and updated the readme. Also I added the summary from tools-cli as the help output. Let me know if there is something else that I can do.

borkdude commented:
Cool! I think the default of `--cache-dir` should be based on `--repo` in the README/help, since when `--repo` changes, the default cache-dir changes.

mynomoto committed:
Better description of the default cache-dir option.

mynomoto commented:
Is this better? If not I'm open to suggestions. 😉

borkdude commented:
Thank you!
